server/configuration-validation.js

```diff
@@ -1,127 +1,195 @@
+      }
+
+      return this
+    },
+
+    validateFilterEntry() {
+      // filterEntry is either undefined or a function
+      if (filterEntry !== undefined && typeof filterEntry !== 'function') {
+        log.error(
+          `The "filterEntry" option of "${collectionName}" should be a function`
+        )
+      } else if (filterEntry !== undefined) {
+        options.filterEntry = filterEntry
+      }
+
+      return this
+    },
+
+    validateMeilisearchSettings() {
+      // Settings is either undefined or an object
+      if (settings !== undefined && !isObject(settings)) {
+        log.error(
+          `The "settings" option of "${collectionName}" should be an object`
+        )
+      } else if (settings !== undefined) {
+        options.settings = settings
+      }
+
+      return this
+    },
+
+    validatePopulateEntryRule() {
+      // PopulateEntry is either undefined or an object/array/string
+      if (
+        populateEntryRule !== undefined &&
+        !isObject(populateEntryRule) &&
+        !Array.isArray(populateEntryRule) &&
+        typeof populateEntryRule !== 'string'
+      ) {
+        log.error(
+          `The "populateEntryRule" option of "${collectionName}" should be an object/array/string`
+        )
+      } else if (populateEntryRule !== undefined) {
+        options.populateEntryRule = populateEntryRule
+      }
+
+      return this
+    },
+
+    validateNoInvalidKeys() {
+      // Keys that should not be present in the configuration
+      Object.keys(excedent).map(key => {
+        log.warn(
+          `The "${key}" option of "${collectionName}" is not a known option`
+        )
+      })
 
-  for (const collection in collections) {
-    config[collection] = validateCollectionConfiguration({
-      configuration: collections[collection],
-      collection: collection,
-    })
+      return this
+    },
+    get() {
+      return options
+    },
   }
 }
 
-function validateCollectionConfiguration({ configuration, collection }) {
-  const validApiFields = [
-    'indexName',
-    'transformEntry',
-    'settings',
-    'filterEntry',
-    'populateEntryRule',
-  ]
+function PluginConfig(configuration) {
+  const log = strapi.log // has to be inside a scope
+  const { apiKey, host, ...collections } = configuration
+  const options = {}
 
-  if (configuration === undefined) {
-    return
-  }
+  return {
+    validateApiKey() {
+      // apiKey is either undefined or a string
+      if (apiKey !== undefined && typeof apiKey !== 'string') {
+        log.error('The "apiKey" option should be a string')
+      } else if (apiKey !== undefined) {
+        options.apiKey = apiKey
+      }
+      return this
+    },
 
-  if (configuration !== undefined && !isObject(configuration)) {
-    strapi.log.error(`The collection "${collection}" should be of type object`)
-    return {}
-  }
+    validateHost() {
+      // // apiKey is either undefined or a none empty string
```

Comment:
```suggestion
      // apiKey is either undefined or a none empty string
```

I think you did not apply the suggestion/did it yourself

Reply:
Forgot to push :/ sorry